src/Sulu/Bundle/MediaBundle/Controller/MediaController.php

```diff
@@ -101,85 +103,102 @@ function ($id) use ($locale, $mediaManager) {
     }
 
     /**
-     * lists all media.
+     * Lists all media.
      *
      * @param Request $request
      *
-     * @return \Symfony\Component\HttpFoundation\Response
+     * @return Response
      */
     public function cgetAction(Request $request)
     {
-        try {
-            $collectionId = $request->get('collection');
-
-            if ($collectionId) {
-                $collection = $this->getCollectionManager()->getById($collectionId, $this->getLocale($request));
-                if ($collection->getType()->getKey() === SystemCollectionManagerInterface::COLLECTION_TYPE) {
-                    $this->getSecurityChecker()->checkPermission(
-                        'sulu.media.system_collections',
-                        PermissionTypes::VIEW
-                    );
-                }
+        $fieldDescriptors = $this->getFieldDescriptors($this->getLocale($request));
+        $listBuilder = $this->getListBuilder($request, $fieldDescriptors);
+        $listBuilder->addSelectField($fieldDescriptors['version']);
+        $listBuilder->addSelectField($fieldDescriptors['name']);
+        $listBuilder->addSelectField($fieldDescriptors['locale']);
+        $listBuilder->addSelectField($fieldDescriptors['id']);
```

Comment:
Ok, the `RestHelper` is used in the `getListBuilder` function, but why do these selects then have to be set here again?